pages/api/cck/auth/register.ts

```diff
@@ -0,0 +1,42 @@
+import { Organization, Person } from '@/components/db-connection';
+import crypto from 'node:crypto';
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { CCKPersonInsert } from '@/components/db-connection/CCKPerson';
+import { Prisma } from '@prisma/client';
+
+/**
+ * Allows the user to register as a CCK volunteer
+ */
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+
+	const reqBody = req.body as CCKPersonInsert;
+
+	if (!reqBody.email || !reqBody.addl_info || !reqBody.addl_info.nickname) {
+		res.status(400).json({ result: "Request body does not contain the requisite information"});
+		return;
+	}
+
+	if (await Person.isPersonInOrg({email: reqBody.email}, "cck")) {
+		res.status(400).json({ error: "User already exists"});
+		return;
+	}
+
+	if (await Organization.hasPersonWithInfoProp({id_ref: "cck"}, "nickname", reqBody.addl_info.nickname)) {
+		res.status(400).json({ error: "User with nickname already exists"});
+		return;
+	}
+
+	const fullPerson: Person.PersonInsert = {
+		email: reqBody.email,
+		totpsecret: crypto.randomBytes(20).toString('hex'),
```

Comment:
It would be more efficient and standard to have the byte length to be in a multiple of 8, I'd recommend choosing 32 for a hex string length of 64 and a hash size space of 1.2×10^77